src/pybind/mgr/cephadm/agent.py

```diff
@@ -719,12 +719,34 @@ def stop(self) -> None:
         self.unsubscribe()
         super().stop()
 
+    def get_data(self) -> Optional[dict]:
+        import io
+        import gzip
+        try:
+            content_encoding = cherrypy.request.headers.get('Content-Encoding', '')
+            remote_ip = cherrypy.request.remote.ip
+            raw_body = cherrypy.request.body.read()
+            if content_encoding == 'gzip':
+                self.mgr.log.info(f">>> Received gzipped payload from {remote_ip}")
+                buf = io.BytesIO(raw_body)
+                with gzip.GzipFile(fileobj=buf) as gz:
+                    decompressed = gz.read()
+                data = json.loads(decompressed.decode('utf-8'))
+            else:
+                self.mgr.log.info(f">>> Received plain payload from {remote_ip}")
+                data = json.loads(raw_body.decode('utf-8'))
+            return data
+        except Exception as e:
+            self.mgr.log.error(f"Failed to read request body: {e}")
+            return None
```

Comment:
Cherrypy has a built-in tool for gzip compression. [This is Dashboard PR](https://github.com/ceph/ceph/pull/24727/files).

Reply:
@epuertat not sure why, but the built-in tool seems to work only for outgoing traffic (responses). I get errors when I try to decompress incoming gzip traffic without doing it manually (as in the actual code).

Reply:
@epuertat I had a discussion on slack with cherrypy maintainer and he confirmed to me that in fact the built-in tool for gzip only works for outgoing traffic (resopnses). So I'm afraid we need to "manually" uncompress the body when processing the response.